.github/workflows/create-newsletter.yaml

```diff
@@ -0,0 +1,45 @@
+name: Create Newsletter
+on:
+  schedule:
+    # Run on the 5th of every month at 12pm
+    - cron: "0 12 5 * *"
+  workflow_dispatch:
+    inputs: {}
+permissions:
+  contents: write
+  issues: write 
+jobs:
+  create-newsletter:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set environment variables
+        run: |
+          echo "NEWSLETTER_MONTH=$(date +'%B')" >> $GITHUB_ENV
+          echo "NEWSLETTER_YEAR=$(date +'%Y')" >> $GITHUB_ENV
+          echo "NEWSLETTER_DEADLINE=$(date -d' 1 month ' '+%m.%Y')" >> $GITHUB_ENV
+          # take the amount of months since feb 2024 and add 50
+          echo "NEWSLETTER_COUNTER=$(( ( ( $(date +'%Y') - 2024 ) * 12 + $(date +'%m') - 2 ) + 50 ))" >> $GITHUB_ENV
+      - name: Create newsletter scaffold
+        run: |
+          cp newsletter-template.md index.md
```

Comment:
Nit: I'd love an external script to run for this, so we can run it locally and verify it's logic, if we ever want to makes changes, or github action is down, or whatever (tests...)

Reply:
You use [act](https://github.com/nektos/act) for running GitHub actions locally

Reply:
@Vrixyz I've extracted it into an own `.sh` now :)